internal/datacoord/compaction_trigger.go

```diff
@@ -685,22 +656,29 @@ func isDeleteRowsTooManySegment(segment *SegmentInfo) bool {
 	return is
 }
 
-func (t *compactionTrigger) ShouldStrictCompactExpiry(fromTs uint64, compactTime *compactTime, signal *compactionSignal, segment *SegmentInfo) bool {
-	if signal != nil && signal.doStrictExpiryCompaction && fromTs <= compactTime.expireTime {
-		log.Info("Trigger strict expiry compaction for segment",
-			zap.Int64("segmentID", segment.GetID()),
-			zap.Int64("collectionID", segment.GetCollectionID()),
-			zap.Int64("partition", segment.GetPartitionID()),
-			zap.String("channel", segment.GetInsertChannel()),
-			zap.Uint64("segment fromTs", fromTs),
-			zap.Uint64("expireTime", compactTime.expireTime),
-		)
-		return true
+func (t *compactionTrigger) ShouldCompactExpiry(fromTs uint64, compactTime *compactTime, segment *SegmentInfo) bool {
```

Comment:
if the expire_time is 100, toleration is 50
then currently, data like 90, 80, 70 are all expired
but the toleration is 50, so the earliesTolerance is 100-50=50
then under such a case, segment will fromts later than 50, like 70, 80 will be kept for the time being
but segment with fromts=30/40 will be submitted to compaction